app/src/main/kotlin/com/wire/android/ui/home/conversations/search/SearchUserViewModel.kt

```diff
@@ -61,7 +66,29 @@ class SearchUserViewModel @Inject constructor(
     @VisibleForTesting
     suspend fun safeSearch(query: String) {
         val (searchTerm, domain) = federatedSearchParser(query)
+        val isHandleSearch = validateUserHandle(searchTerm.removeQueryPrefix()) is ValidateUserHandleResult.Valid
 
+        if (isHandleSearch) {
+            searchByHandle(searchTerm, domain)
+        } else {
+            searchByName(searchTerm, domain)
+        }
+    }
+
+    private suspend fun searchByHandle(searchTerm: String, domain: String?) {
+        searchByHandleUseCase(
+            searchTerm,
+            excludingConversation = addMembersSearchNavArgs?.conversationId,
+            customDomain = domain
+        ).also { userSearchEntities ->
+            state = state.copy(
+                contactsResult = userSearchEntities.connected.map(contactMapper::fromSearchUserResult).toImmutableList(),
+                publicResult = userSearchEntities.notConnected.map(contactMapper::fromSearchUserResult).toImmutableList()
+            )
+        }
+    }
+
+    private suspend fun searchByName(searchTerm: String, domain: String?) {
```

Comment:
I don't get: why not move whole this logic into kalium UseCase? 
We could have update `SearchUsersUseCase` to validate if it should search by name or by handler. AND in that case the name "SearchUsersUseCase" would be logical as IMO right now it should be named "SearchByNameUseCase".

Reply:
If the search term begins with "@", it indicates a handle search, while any other start indicates a name search; this behavior is specific to the Android app. However, for the three other Kalium consumers who favor a clearer API, the SearchByHandleUseCase will be dedicated solely to handle searches, and similarly, there will be a distinct use case for name searches. 

We need to clearly define the responsibilities of the app versus those of Kalium. I would prefer to have an explicit API rather than use cases that are easier to use but contain internal logic that isn't explicitly clear to consumers.